hypertrace-graphql-entity-schema/src/main/java/org/hypertrace/graphql/entity/request/EdgeRequestBuilder.java

```diff
@@ -165,6 +177,17 @@ private Single<AttributeRequest> getNeighborIdAttribute(
     }
   }
 
+  private List<FilterArgument> getFilters(Set<SelectedField> selectedFields) {
+    return selectedFields.stream()
+        .map(
```

Comment:
referred to in other comment, but this is combining the filter arguments from each independent edge request (each field here is one edge request like `incomingEdges(type: X)`).

Reply:
This I have already discussed with @skjindal93 . With this PR we are two limitations in the case of using edge filters,
1. If using edge filters, we can only have one entityType incoming/outgoing edge
2. In the case of different filters for each selected field. They will be merged into one AND filter.

we are introducing this to solve the deliverable first. Immediately after this we will make the changes to remove above two limitations. Currently, no one is using multiple entity types interactions, so it won't break anything. also old behaviour is still intact.

Reply:
The reason we didnt solve this is because it will need big change in gateway service where we will have to make interactions request as an array of request rather than one single filter which it is right now. https://github.com/hypertrace/gateway-service/blob/main/gateway-service-api/src/main/proto/org/hypertrace/gateway/service/v1/entities.proto#L41 this is the filter i am referring to

Reply:
I'm not sure I agree here, but we can discuss further.

> Currently, no one is using multiple entity types interactions, so it won't break anything

Confused here - the main application flow uses multiple entity types? It shows services with downstream services and downstream backends.


> The reason we didnt solve this is because it will need big change in gateway service where we will have to make interactions request as an array of request rather than one single filter which it is right now.

We already handle multiple entity types through a filter, is this different? You'd need to just change how you build the filter to OR across selection types and then AND for each selection type.

Reply:
> Confused here - the main application flow uses multiple entity types? It shows services with downstream services and downstream backends.

It won't break the current workflows. Those api will work. These limitations only come into the picture when filters are being used.

Reply:
We don't want to leave an api in this broken state though - so if this is going to impact how we implement things, we should be considering that now.

Reply:
We were just planning to do it in iterations. Have a basic application flow working with limitations (not breaking the existing application flow), and then refactor it next to make it perfect. (because, refactoring would take considerable amount of time otherwise, causing delays for Third Party APIs)

We are definitely planning to pick up the refactor as the next task

Reply:
It shouldn't require a big refactor though - that's the part that's confusing me. The refactor would just be in the new code that's being written in these 2 PRs, right? I can definitely be mistaken, but aren't we talking a handful of lines (since we already have the infrastructure to separate out the requests then merge them for the existing code)? 

If we choose not to however, can we at least have this error out if we receive an unsupported request? My bigger concern is that certain calls will just result in success but have inaccurate results (because filters get merged that shouldn't be).

Reply:
> We already handle multiple entity types through a filter, is this different? You'd need to just change how you build the filter to OR across selection types and then AND for each selection type.

This won't work actually if I am understanding it correctly. Consider this for an example, let's say we query for two edges like this
```
incomingEdge (scope: SERVICE, filter: {dataType in "dataTypeIds1"})
incomingEdge (scope: API, filter: {dataType in "dataTypeIds2"} }
```

Now the interaction request to the gateway service will be like this
```
filter AND [
{ fromEntityType in SERVICE, API}
{ dataType in dataTypeIds1 } OR { dataType in dataTypeIds2 }
]
```
Now this will return wrong results for both service and api. Since each filter corresponding to the edge is specific to that interaction we cannot do it in a single query. We will need to change the API contract and make gateway service to run each edge query separately. For now, we have made the code error out in case of above-mentioned limitations

Reply:
> Now this will return wrong results for both service and api. Since each filter corresponding to the edge is specific to that interaction we cannot do it in a single query

I was actually suggesting:
```
filter OR [
{{ fromEntityType = SERVICE} AND { dataType in dataTypeIds1 }}
{{ fromEntityType = API} AND { dataType in dataTypeIds2 }}
]
```
Would that be problematic?